### ISSUE TITLE
When loading ipfs.io with `www` some things are broken

### ISSUE THREAD

**Issue by 2color** [ipfs/ipfs-website#141]
## Background
Currently, the IPFS website is available over two domains:
- https://www.ipfs.io/
- https://ipfs.io/

## Problem

The two domains are served by different servers that serve different assets as indicated by DNS resolution:

```
➜  dig www.ipfs.io

; <<>> DiG 9.10.6 <<>> www.ipfs.io
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 60556
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;www.ipfs.io.			IN	A

;; ANSWER SECTION:
www.ipfs.io.		38	IN	CNAME	ffb10528e9a368494097.b-cdn.net.
ffb10528e9a368494097.b-cdn.net.	13 IN	A	138.199.37.226

;; Query time: 11 msec
;; SERVER: 192.168.178.1#53(192.168.178.1)
;; WHEN: Wed Jul 20 16:15:18 CEST 2022
;; MSG SIZE  rcvd: 100

➜  dig ipfs.io

; <<>> DiG 9.10.6 <<>> ipfs.io
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 60220
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;ipfs.io.			IN	A

;; ANSWER SECTION:
ipfs.io.		2211	IN	A	209.94.90.1

;; Query time: 11 msec
;; SERVER: 192.168.178.1#53(192.168.178.1)
;; WHEN: Wed Jul 20 16:15:21 CEST 2022
;; MSG SIZE  rcvd: 52

```

The problem is that the two domains return a different URL for the 3 JS files that don't load on the apex domain (without `www`).

The working version has these JS scripts in the payload
- https://www.ipfs.io/_nuxt/static/1656326872/state.js
- https://www.ipfs.io/_nuxt/static/1656326872/payload.js
- https://www.ipfs.io/_nuxt/static/1656326872/manifest.js

While the non-functioning apex version has the following scripts in the payload
- https://ipfs.io/_nuxt/static/1651855547/state.js
- https://ipfs.io/_nuxt/static/1651855547/payload.js
- https://ipfs.io/_nuxt/static/1651855547/manifest.js  

<img width="1070" alt="Screen Shot 2022-07-18 at 6 41 19 PM" src="https://user-images.githubusercontent.com/1992255/179560778-11fd68af-e074-4d9e-8c84-609f540bfc65.png">

Moreover, this causes the animation to not load:
<img width="1610" alt="Screen Shot 2022-07-18 at 6 44 31 PM" src="https://user-images.githubusercontent.com/1992255/179561233-054c3026-5f2a-4504-8e87-32743ef00c17.png">

And for the News and More section to be broken:

<img width="1671" alt="Screen Shot 2022-07-18 at 6 45 16 PM" src="https://user-images.githubusercontent.com/1992255/179561372-03b162bd-878c-44bb-998f-314bb1effdc9.png">

## Possible solutions

- Redirect visitors to `ipfs.io` -> `www.ipfs.io` if the path doesn't contain `/ipfs` which is used for the public gateway.
- Fix the reverse proxy that handles requests to `https://ipfs.io/_nuxt/static/1651855547/state.js` so that it handles it the same way as `https://www.ipfs.io/_nuxt/static/1651855547/state.js`

## Related 

- [ ] Update the canonical URL to www.ipfs.io

**Comment by lidel** [comment]
I believe we need to park this until we have the new domain  situation sorted in https://github.com/protocol/bifrost-infra/issues/178#issuecomment-1147558737 

Then, we will normalize the URL using the code present in `layouts/default.vue` from https://github.com/ipfs/ipfs-website/pull/133 (making sure users end up on canonical URL)

**Comment by 2color** [comment]
> when it comes to writing normalization,

@lidel What do you mean by writing normalization in this context?

**Comment by lidel** [comment]
ah, yes, I meant "redirecting the user to the canonical domain name"  (used "normalization" because it involves a bit more work than just returning HTTP 301 redirect – see example in https://github.com/ipfs/ipfs-website/pull/133)

**Comment by lidel** [comment]
Quick update: debugged this together with @thattommyhall, and we've found potential cause: stale cache on `/ipns/`. 
Fix in https://github.com/ipfs/kubo/pull/9132, will be included in Kubo 0.14.0 (should be released by EOW).

**Comment by 2color** [comment]
This seems to have been resolved as part of the move to ipfs.tech. 

I also noticed that we have client side url normalisation in place, ensuring that visiting `www.ipfs.tech` redirects to `ipfs.tech` 💯
